pkg/network/cluster_config.go

```diff
@@ -39,15 +41,20 @@ func ValidateClusterConfig(clusterConfig configv1.NetworkSpec) error {
 		if err != nil {
 			return errors.Errorf("could not parse spec.clusterNetwork %s", cnet.CIDR)
 		}
-		ones, bits := cidr.Mask.Size()
-		// The comparison is inverted; smaller number is larger block
-		if cnet.HostPrefix < uint32(ones) {
-			return errors.Errorf("hostPrefix %d is larger than its cidr %s",
-				cnet.HostPrefix, cnet.CIDR)
-		}
-		if int(cnet.HostPrefix) > bits-2 {
-			return errors.Errorf("hostPrefix %d is too small, must be a /%d or larger",
-				cnet.HostPrefix, bits-2)
+		//ignore hostPrefix if the plugin does not use it
+		for _, plugin := range pluginsUsingHostPrefix {
+			if plugin == clusterConfig.NetworkType {
```

Comment:
Looping over `pluginsUsingHostPrefix` like this is a weird way to check this. You should make a helper function `networkTypeUsesHostPrefix` or something, and just check that from here.

Reply:
Also, we should not allow a "bad" HostPrefix even if it's not being used. We should allow it to be `0` if the plugin doesn't use HostPrefix, but if it's not 0, then it should still be required to be valid.

Reply:
> Also, we should not allow a "bad" HostPrefix even if it's not being used. We should allow it to be `0` if the plugin doesn't use HostPrefix, but if it's not 0, then it should still be required to be valid.

hmm ok yea so that means the plugins that don't use this field can set it to zero, but if they haven't set it to 0 they have to comply with our definition of a valid hostPrefix. That sounds reasonable and you are right - allowing to set anything in that field and not check is bad UI. I'll also try to add a document note somewhere for reference.